src/lib.rs

```diff
@@ -192,8 +125,8 @@ where
     where
         A: Readable,
     {
-        // UNSAFE: Safe, as we know that our internal value exists.
-        unsafe { ptr::read_volatile(&*self.reference) }
+        // UNSAFE: Safe, as ... TODO
```

Comment:
The `voladdress` crate actually manages to use some type bound shenanigans to make the wrapping method safe or not, see this part: https://github.com/rust-console/voladdress/blob/main/src/plain_voladdress.rs#L141-L190

This way, the safety requirements fall to the declaration site, not on the read and write methods themselves.

Reply:
That's an interesting approach! I like that you incorporate the access permissions in it directly (i.e. using `()` for "no access" ). The drawback is that it makes the type signatures more complex.

For this crate, it might work to integrate these properties into the `access` parameter. I'll look into it.

Reply:
I pushed a prototype for `unsafe` read/write support using a new `access::Custom` type in https://github.com/rust-osdev/volatile/pull/22/commits/682dd7092e8fc0cc7226db5cfaf4208d09530321.